configure

```diff
@@ -773,7 +773,13 @@ def glob_to_var(dir_base, dir_sub):
     (path, dirs, files) = ent
     for file in files:
       if file.endswith('.cpp') or file.endswith('.c') or file.endswith('.h'):
-        list.append('%s/%s' % (dir_sub, file))
+        srcfile = '%s/%s' % (dir_sub, file)
+        if patch_dir:
+          patchfile = '%s/%s/%s' % (dir_base, patch_dir, file)
+          if os.path.isfile(patchfile):
+            srcfile = '../../%s' % patchfile
```

Comment:
[reposting as it looks like my previous comment was lost]

`../..` is hardcoded and it wouldn't work with all possible inputs or if the directory structure was changed. If I understand correctly, the paths are assumed to be relative to the location of icu-generic.gyp (which is in tools/icu), and the same is passed to this function as dir_base. Since patch_dir is also given relative to dir_base, you could just do:
`srcfile = '%s/%s' % (patch_dir, file)`